ckan/logic/action/get.py

```diff
@@ -338,12 +339,16 @@ def _group_or_org_list(
     all_fields = asbool(data_dict.get('all_fields', None))
 
     if all_fields:
+        if set_max_limit:
```

Comment:
@TomeCirun @wardi I was the one who introduced the two `group_list` calls in the group index view function (then controller) in https://github.com/ckan/ckan/pull/2554, believe it or not to improve performance :)

The reasoning was that for the index page we need the group details for all items in that page (`all_fields=True`) but also we need the total number of groups to be able to paginate properly (2on call with `all_fields=False`). I know it's not elegant but going back to a single call with `all_fields=True` without limit will reduce performance for sure.

@wardi can you think of an alternative approach?